Fix msal test app after merging with the fadi/release/5.4.0

Yuki-YuXin opened the pull request:
Fix update according to the following team feedback:
**Batch2**
- [hero scenario 6, use case 2.2.1] Use email and OTP to get token and sign in
getParcelable(Constants.STATE) 
- [use case 2.2.6] Ability to provide scope to control auth strength of the token
Update the scope of profile to User.Read 
Update the test app EmailSignInSignUp page: signIn(scopes = User.Read)
- [use case 1.2.2] User is not registered with given email
Update test app EmailPasswordSignInSignUp page: displayDialog("Unexpected result", actionResult.errorMessage) in sign in function
- [use case 1.2.6] Ability to provide scope to control auth strength of the token
Update the test app EmailPasswordSignInSignUp page: signIn(scopes = User.Read)
Update test app EmailPasswordSignInSignUp page: displayDialog("Unexpected result", actionResult.errorMessage) in sign in function
- [hero scenario 2, use case 2.1.2] Signup user with custom attributes with verify OTP as last step
Update the test app SignUpCode page: currentState = (bundle?.getParcelable(Constants.STATE) as? SignUpCodeRequiredState)!!
- [hero scenario 2, use case 2.1.2] Signup user with custom attributes with verify OTP as last step
Update the test app SignUpCode page: currentState = (bundle?.getParcelable(Constants.STATE) as? SignUpCodeRequiredState)!!
- [hero scenario 2, use case 2.1.2] Signup user with custom attributes with verify OTP as last step

- [hero scenario 3, use case 2.1.3] Verify email OTP first and then collect custom attributes
Update the test app SignUpAttributes page: currentState = (bundle?.getParcelable(Constants.STATE) as? SignUpAttributesRequiredState)!!
SignUpAttributesUI disorder

- [use case 1.1.4] Verify email address using email OTP and then set password 
In SignUpCodeFragment modify to currentState = (bundle?.getParcelable(Constants.STATE) as? SignUpCodeRequiredState)!!
In SignUpPasswordFragment modify to currentState = (bundle?.getParcelable(Constants.STATE) as? SignUpPasswordRequiredState)!!


**Batch1**
- [use case 1.2.2] User is not registered with given email
No error is popped up in UI (Probably this is a bug of the test app)
- [use case 2.2.5] Resend email OTP
App crashing
- [use case 1.2.3] Password is incorrect
No error is popped up in UI (Probably this is a bug of the test app)
- [hero scenario 3, use case 2.1.3] Verify email OTP first and then collect custom attributes
When submitting attributes (had to hard-code some), the test application doesn't return to the sign in / sign up screen
- Authenticate user (get token) through web browser, retrieve tokens through native auth SDK interface (by setting forceRefresh=true): 
EmailSignInSignUpFragment does not have logic for handling redirects
 - [hero scenario 6, use case 2.2.1] Use email and OTP to get token and sign in

rmccahill commented on testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailSignInSignUpFragment.kt:
```diff
@@ -108,66 +114,90 @@ class EmailSignInSignUpFragment : Fragment() {
+            val actionResult = authClient.signIn(
+                username = email
+            )
+            System.out.println("2")
```
Looks like some left over debugging here, same on line 119

Yuki-YuXin replied:
Sure. I will remove this line. We may not need test app any longer if the E2E tests are done.

rmccahill replied:
I think it will still be useful for manually debugging flows, and for checking incidents for on call engineers as well

rmccahill commented on msal/versioning/version.properties:
```diff
@@ -1,3 +1,3 @@
 #Wed Aug 01 15:24:11 PDT 2018
-versionName=5.3.1
+versionName=5.4.0
```
Should these versions be updated as part of these changes? Same applies for the changes in `msalautomationapp/build.gradle`, and `msal/build.gradle`

Yuki-YuXin replied:
I think the changes is because of the merging with the dev or fadi/release/branch

fadidurah committed:
release updates

fadidurah committed:
changelog

Yuki-YuXin committed:
Fix test app issues

algo-1 committed:
Update the msal test app for Native Auth (#2143)

Fix the MSAL test app to support custom attribute test cases without the
need to modify the source code.

[ADO Ticket
(2979358)](https://dev.azure.com/IdentityDivision/Engineering/_boards/board/t/DevExDub%20-B2C/Backlog%20items/?workitem=2979358)

---------

Co-authored-by: Harold Karibiye <tkaribiye@microsoft.com>

Yuki-YuXin committed:
Merge branch 'dev' into sammy/fix-msal-test-app

# Conflicts:
#	common

SammyO commented on changelog:
```diff
@@ -3,6 +3,10 @@ MSAL Wiki : https://github.com/AzureAD/microsoft-authentication-library-for-andr
 
+Version 5.4.1-RC1
+---------
+- [PATCH] Update common @17.6.0-RC1
```
This doesn't belong here. MSAL changelog is for changes that happened in MSAL.

SammyO commented on msal/build.gradle:
```diff
@@ -188,12 +188,12 @@ task sourcesJar(type: Jar) {
 }
 // Used for testfixtures
-def common4jVersion = "1.0.+"
+def common4jVersion = "14.6.0-RC1"
```
I don't think this is right either? Neither the above.

Yuki-YuXin replied:
That might because I merged release branch into the test app branch in order to have both test environment. I will check the content in the dev branch and revert it back.

SammyO commented on msal/versioning/version.properties:
```diff
@@ -1,3 +1,3 @@
 #Wed Aug 01 15:24:11 PDT 2018
-versionName=5.4.0
+versionName=5.4.1-RC1
```
This is not right either.

SammyO commented on testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailAttributeSignUpFragment.kt:
```diff
@@ -132,7 +132,7 @@ class EmailAttributeSignUpFragment : Fragment() {
                 val actionResult = authClient.signUp(
                     username = email,
-                    password = password,
+//                    password = password,
```
Please remove commented out code.

SammyO commented on testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailPasswordSignInSignUpFragment.kt:
```diff
@@ -120,7 +120,8 @@ class EmailPasswordSignInSignUpFragment : Fragment() {
                     username = email,
-                    password = password
+                    password = password,
+                    scopes = listOf("User.Read")
```
I'm not against it, but is it necessary?

Yuki-YuXin replied:
[use case 2.2.6] Ability to provide scope to control auth strength of the token
 In this test case, we need to modify the code, providing  scopes = listOf("User.Read") . If it had been there all along, the testers wouldn't have needed any modifications

SammyO replied:
Got it.

SammyO commented on testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailPasswordSignInSignUpFragment.kt:
```diff
@@ -279,6 +283,8 @@ class EmailPasswordSignInSignUpFragment : Fragment() {
                 binding.resultAccessToken.text =
                     getString(R.string.result_access_token_text) + accessToken
 
+                Log.d("AccessToken", accessToken)
```
Why is this needed? If not, let's not log this. This is sensitive data and we should be setting an example.

SammyO commented on testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailSignInSignUpFragment.kt:
```diff
@@ -112,7 +114,8 @@ class EmailSignInSignUpFragment : Fragment() {
                 val actionResult = authClient.signIn(
-                    username = email
+                    username = email,
+                    scopes = listOf("User.Read")
```
Same here.

SammyO commented on testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/nativeauth/EmailSignInSignUpFragment.kt:
```diff
@@ -250,6 +253,8 @@ class EmailSignInSignUpFragment : Fragment() {
                 binding.resultAccessToken.text =
                     getString(R.string.result_access_token_text) + accessToken
 
+                Log.d("AccessToken", accessToken)
```
Same here

Yuki-YuXin committed:
Merge branch 'dev' into sammy/fix-msal-test-app

# Conflicts:
#	changelog
#	msal/versioning/version.properties

Yuki-YuXin committed:
Deal with batch 2 feedback

Yuki-YuXin committed:
Merge branch 'dev' into sammy/fix-msal-test-app

Yuki-YuXin committed:
Apply dev version

Yuki-YuXin committed:
Update the UserAttributes.Builder() and Reuse Email password attributes UI for email OTP attributes case

Yuki-YuXin committed:
Update the UserAttributes.Builder() and Reuse Email password attributes UI for email OTP attributes case

Yuki-YuXin committed:
Merge branch 'dev' into sammy/fix-msal-test-app
